app/controllers/home_controller.rb

```diff
@@ -47,6 +47,29 @@ def dashboard
     end
   end
 
+  def dashboard2
+    # The new dashboard displays the blog and topics list
+    if current_user
+      @blog = Tag.find_nodes_by_type('blog', 'note', 1).first
+      # Topics a user has subscribed to
+      subscibed_tags = TagSelection.includes(:tag)
```

Comment:
This is looking like a great start, and i like that you're starting with just one query. I bet there are other pages we can look at to find fast standard ways to get this info, too! Here, I think we could use `user.subscriptions` -- 

https://github.com/publiclab/plots2/blob/6505f539cabcb9aa4eb9fc5c00af6cf9584351a2/app/models/user.rb#L191-L196

Then, i search for where this is used in code, to see examples:

https://github.com/publiclab/plots2/search?q=user.subscriptions - and indeed we see a nice example, which also includes the `tag` table, so we get the whole record we need. (`tag_selection` is just a join table, i think!)

```ruby
    <% current_user.subscriptions(:tag).includes(:tag).each do |subscription| %>
```

Actually i wish we had a unit test, as that's usually a great guide. So I'll check there too - aww, looks like https://github.com/publiclab/plots2/blob/main/test/unit/user_test.rb has no example for `user.subscriptions` -- that's too bad! Maybe we make a to-do for ourselves to add a simple unit test there.

As to many of the queries below, the template we'll be using actually does a lot of that work -- it expects a `@tags` variable and will build the whole listing from that:

https://github.com/publiclab/plots2/blob/main/app/views/tag/_topicCard.html.erb

So, if we do this, we might get very close!

```suggestion
      @tags = current_user.subscriptions(:tag).includes(:tag)
```

What do you think?

Reply:
This totally makes sense, thanks @jywarren, I'll do some further research and update the PR..

Reply:
Hi, @jywarren I've added the subscriptions test as well as the query.